clustermesh/clustermesh.go

```diff
@@ -797,6 +805,10 @@ func (k *K8sClusterMesh) patchConfig(ctx context.Context, client k8sClusterMeshI
 }
 
 func (k *K8sClusterMesh) Connect(ctx context.Context) error {
+	if err := k.Validate(ctx, true); err != nil {
```

Comment:
This is not enough because `enable` is performed on both clusters and `connect` is only done on one of the clusters. We need to find something better.

Reply:
We extract access information from both clusters right after the above lines, will move the cluster sanity check there.

Reply:
As a bonus we can also check that the clusters actually have different names and IDs :-)